lib/winapi/winbase.h

```diff
@@ -45,6 +45,16 @@ typedef struct _INIT_ONCE
 #define INIT_ONCE_INIT_FAILED 0x00000004UL
 typedef BOOL CALLBACK PINIT_ONCE_FN(PINIT_ONCE, PVOID, PVOID *);
 
+typedef struct  _SRWLOCK
+{
+    DWORD_PTR Ptr;
+    // highest bit: reader lock (0=free, 1=locked)
+    // second-highest bit: global lock (0=free, 1=locked)
+    // remainder: blocking readers
```

Comment:
nit: Consider defining masks for this in the file where this is implemented, something like `_SRWLOCK_READER_LOCKED`, `_SRWLOCK_GLOBAL_LOCK`, `_SRWLOCK_READERS`

I had a lot of trouble reading the code with all the numeric `AND` masks - I wouldn't be surprised if there are bugs as a result.

Reply:
Did that. I even found a bug where I copied some piece of code while forgetting to adjust the mask.